processor/worker.go

```diff
@@ -5,41 +5,163 @@ import (
 	"sync"
 	"time"
 
+	"golang.org/x/sync/errgroup"
+
 	"github.com/rudderlabs/rudder-go-kit/logger"
 	"github.com/rudderlabs/rudder-server/jobsdb"
 	"github.com/rudderlabs/rudder-server/rruntime"
 	"github.com/rudderlabs/rudder-server/services/rsources"
 	"github.com/rudderlabs/rudder-server/utils/misc"
 )
 
-// newProcessorWorker creates a new worker
+type worker struct {
+	partition string
+	workers   []*partitionWorker
+	logger    logger.Logger
+	stats     *processorStats
+	handle    workerHandle
+	g         *errgroup.Group
+}
+
+// newProcessorWorker creates a new worker for the specified partition
 func newProcessorWorker(partition string, h workerHandle) *worker {
 	w := &worker{
+		partition: partition,
+		logger:    h.logger().Child(partition),
+		stats:     h.stats(),
+		handle:    h,
+	}
+	w.g, _ = errgroup.WithContext(context.Background())
+
+	// Create workers for each partition
+	numPartitions := h.config().numPartitions
+	w.workers = make([]*partitionWorker, numPartitions)
+	for i := 0; i < numPartitions; i++ {
+		w.workers[i] = newWorker(partition, h)
+	}
+
+	return w
+}
+
+// Work processes jobs for the specified partition
+// Returns true if work was done, false otherwise
+func (w *worker) Work() bool {
+	// If pipelining is disabled, use the legacy job handling path
+	if !w.handle.config().enablePipelining {
+		return w.handle.handlePendingGatewayJobs(w.partition)
+	}
+
+	// Get jobs for this partition
+	jobs := w.handle.getJobs(w.partition)
+	start := time.Now()
```

Comment:
We should start counting before getting jobs, as it was earlier. Was this change intentional?

```suggestion
	start := time.Now()
	// Get jobs for this partition
	jobs := w.handle.getJobs(w.partition)
```